### PR TITLE
ITNFA-21: added local & national search

### PR DESCRIPTION
**Pull request by pawelcieszko**
Haiti server/Bioplugin is not working so I couldn't test it

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java [hunk 1]
```diff
@@ -107,8 +108,16 @@ public BiometricSubject updateSubjectId(String oldId, String newId) {
 
     @Override
     public List<BiometricMatch> search() {
-        M2SysCaptureResponse capture = scanDoubleFingers();
-        return searchService.search(capture, localBioServerClient);
+        M2SysCaptureResponse fingerScan = scanDoubleFingers();
+        List<BiometricMatch> results = searchService.search(fingerScan, localBioServerClient);
+        if (CollectionUtils.isEmpty(results)) {
+            BiometricMatch nationalResult = searchService.findMostAdequate(fingerScan, nationalBioServerClient);
```
Comment (pgesek) [inline api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java]: why can't we use search here?

Moreover - is the patient imported from the Mpi if this is a national hit? Or will that be a separate commit?

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java [hunk 2]
```diff
@@ -107,8 +108,16 @@ public BiometricSubject updateSubjectId(String oldId, String newId) {
...
+            if (nationalResult != null) {
+                registrationService.fetchFromNational(new BiometricSubject(nationalResult.getSubjectId()), fingerScan);
```
Comment (pgesek) [inline api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java]: why is the registrationService doing a search? Registration service should be only responsible for registration

Comment (pgesek) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java]: Is this the MPI fetch? Can we rename to fetchFromMpiByNationalFpId ?

**Review by pgesek — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java (hunk 2 above)
Comment (dserkowski) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java]: What about naming it fetchFromMpiByNationalBiometricSubject?

Comment (pawelcieszko) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java]: we need only subjectId but it's a string

Comment (pgesek) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java]: NationalFpId is shorter and is the same thing as nationalBiometricSubject -both work for me

Comment (areklalo) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java]: I think we should use the name with word EMPI not an MPI.

Comment (pawelcieszko) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java]: registrationService registers fingerprints and fetches patient (using registration core)

Comment (pgesek) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java]: OpenEMPI is an implementation of the MPI, we can have different mpis

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java [hunk 3]
```diff
@@ -59,7 +60,7 @@ public BiometricSubject enroll(BiometricSubject subject) {
 
         if (!fingerScanStatus.isRegisteredLocally()) {
             if (fingerScanStatus.isRegisteredNationally()) {
-                registrationService.fetchFromNational(fingerScanStatus.getNationalBiometricSubject(), capture);
+                registrationService.fetchFromNationalById(fingerScanStatus.getNationalBiometricSubject(), capture);
```
Comment (pgesek) [inline api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java]: this is even worse than before

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java (hunk 2 above)
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java]: Okay, good to know.

**Comment by pgesek** [comment]
@pawelcieszko you need to resolve conflicts

**Review by dserkowski — APPROVED** [review]
